adapters/repos/db/shard_read.go

```diff
@@ -271,6 +271,9 @@ func (s *Shard) getIndexQueue(targetVector string) (*IndexQueue, error) {
 		}
 		return queue, nil
 	}
+	if targetVector != "" {
+		return nil, fmt.Errorf("index queue: target vector not found: %q", targetVector)
```

Comment:
this method was returning a queue for `targetVector == ""` if there were no targetVectors (named vectors configured, see line 264), now this method changed to handle named vectors, there's no fallback to returning a queue when there's no target vector defined, is that OK?

Reply:
Yes, I figured it's safer to return an error when we are calling this method expecting a named vector when there are none.
Before this change, we could potentially end up pushing to the wrong queue.

Reply:
ok, then if this method is meant to be used only with target vectors, bc now with this check in place we will never reach line 227 (`return s.queue, nil`) yes?

Reply:
> we will never reach line 227 (return s.queue, nil) yes?

Actually not exactly, this method is there to simplify getting the right queue depending on the configuration of the shard: here's an example:

```go
// Example 1: no named vectors
s.getIndexQueue("") // valid call: this reaches l.277

// Example 2: no named vectors
s.getIndexQueue("foo") // invalid call: this reaches l.275

// Example 3: named vectors
s.getIndexQueue("") // invalid call: this reaches l.266

// Example 4: named vectors
s.getIndexQueue("foo") // valid call: this reaches l.272
```

Does that make sense?

Reply:
yeah, now I see it